### DISCUSSION TITLE
Default Browser Icon

### DISCUSSION THREAD

**Discussion by Christoph1974** [nref/BrowseRouter#5] [General]
Hi Slater1, I like the tool very much, but as with browser selector from Dan, there is a little cosmetic issue. If I register "browserouter" as the default browser in windows, all the shortcuts on my desktop show the default icon for "browserselector/browserouter" which is confusing. Is there a way that you would set the default icon of the default browser that is set in the config INI for all desktop shortcuts?

I know this is cosmetics but still confusing...

thanks
Christoph

**Comment by nref** [comment]
Hey Christoph, and thanks for the feedback.

I understand that the BrowseRouter icon appearing for web shortcuts instead of the browser that opens could be confusing.

![image](https://user-images.githubusercontent.com/2089725/232488925-867692fa-8231-4aaf-ab59-f3666f2f83dd.png)

Looking just at this icon, you have no idea which browser will open.

In summary, I think your solution is possible, but I don't want to do it.

Conceivably we could change the icon, which would entail:

1. find the exe of the default real browser on the filesystem (say Firefox) 
2. copy its icon to a new copy of BrowseRouter.exe
3. Clear explorer.exe icon cache
4. kill and restart explorer.exe

Copying icons seems shady. From a UX perspective the icon uniquely identifies to the user which app will run. Copying Firefox's icon seems like an impersonation that I know people won't appreciate.

Moreover, copying the icon can create a different confusion, where the user double-clicks a Firefox icon and due to an unexpected configuration, e.g. Edge launches.

To me, the BrowseRouter icon signals to the user who is aware of BrowserRouter's presence on the system that their double-click will be triaged and the appropriate browser will be launched.

I wonder if we can introduce a different icon, or another design element, such as a splash or Windows tray notification. I'm open to suggestions.

**Comment by Christoph1974** [comment]
Hi Doug

Thanks for your reply. I assumed that it wouldn’t be a proper solution to address this. There are some other constraints, we have seen users keep on clicking the “Chrome is not your default browser” when they use their browser while your solution is the “default” browser. So they revert back to their “normal” browser and this typically kills the use case. Anyhow, many thanks for your assistance.

Kind regards,
Christoph







Christoph Schoch
Geschäftsführer
TwinCap First AG
t: +41 44 666 50 50
 | d: +41 44 520 36 10
e: ***@***.*** | w: www.twincapfirst.ch
a: 
Favreweg 1, 8304 
Wallisellen

**Reply by nref** [reply]
Do let me know what you think we could do to improve the UX!

Regarding the default browser check. That's understandably frustrating. That check can be disabled so that your users never even get asked to reset the browser. There is a different way to do it if Chrome is an enterprise vs single-computer install. This page has info on both.

https://community.spiceworks.com/topic/1700664-disable-chrome-asking-to-be-default-browser-via-group-policy-or-otherwise

**Comment by Christoph1974** [comment]
Hi Doug

Thanks, yep – just thinking what we could do about the icons, good hint with the default browser policy actually. Maybe it would be beneficial if we could include a path to a custom “icon” in the ini file with a custom path, that would allow to use a familiar icon or to “clone” the icon of the default browser, so users wouldn’t even “realize” it – typically corporate IT has set a default anyway….

Thanks
Christoph



Christoph Schoch
Geschäftsführer
TwinCap First AG
t: +41 44 666 50 50
 | d: +41 44 520 36 10
e: ***@***.*** | w: www.twincapfirst.ch
a: 
Favreweg 1, 8304 
Wallisellen

**Reply by nref** [reply]
Here are two ideas
1. Have one of your devs `git clone` this project, change the icon to what you desire, e.g. Chrome, and make a build just for your uses
2. Download a took like [this one](http://angusj.com/resourcehacker/) (no affiliation, use at your own risk) which can reportedly change the icon of any exe.

**Comment by Christoph1974** [comment]
HI Doug

OK, thanks will go for number 1 in this case.

Kr
Christoph



Christoph Schoch
Geschäftsführer
TwinCap First AG
t: +41 44 666 50 50
 | d: +41 44 520 36 10
e: ***@***.*** | w: www.twincapfirst.ch
a: 
Favreweg 1, 8304 
Wallisellen
